tiledb/sm/filesystem/azure.cc

```diff
@@ -179,38 +183,40 @@ Status Azure::init(const Config& config, ThreadPool* const thread_pool) {
         });
   });
 
-  ::Azure::Storage::Blobs::BlobClientOptions options;
-  options.Retry.MaxRetries = max_retries;
-  options.Retry.RetryDelay = retry_delay_;
-  options.Retry.MaxRetryDelay = max_retry_delay;
-
-  SSLConfig ssl_cfg = SSLConfig(config);
-  options.Transport.Transport = create_transport(ssl_cfg);
-
-  // Construct the Azure SDK blob service client.
-  // We pass a shared key if it was specified.
-  if (!account_key.empty()) {
-    client_ =
-        tdb_unique_ptr<::Azure::Storage::Blobs::BlobServiceClient>(tdb_new(
-            ::Azure::Storage::Blobs::BlobServiceClient,
-            blob_endpoint,
-            make_shared<::Azure::Storage::StorageSharedKeyCredential>(
-                HERE(), account_name, account_key),
-            options));
-  } else {
-    client_ =
-        tdb_unique_ptr<::Azure::Storage::Blobs::BlobServiceClient>(tdb_new(
-            ::Azure::Storage::Blobs::BlobServiceClient,
-            blob_endpoint,
-            options));
+  std::lock_guard<std::mutex> lck(client_init_mtx_);
+
+  if (!client_) {
+    ::Azure::Storage::Blobs::BlobClientOptions options;
+    options.Retry.MaxRetries = azure.max_retries_;
```

Comment:
This is set in the azure client from the passed config and used only here. get() should just take in the config at this point and we shouldn't store the settings as members of the Azure class anymore.

Reply:
Moved configuration options to a separate `AzureParameters` class, which is what we now pass to `AzureClientSingleton`.